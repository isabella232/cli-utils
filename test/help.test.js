const { getHelp } = require('../lib/help');

const flagSchema = require('./fixtures/flags.json');

describe('help', () => {
  test('getHelp', () => {
    expect(getHelp).toBeDefined();
    expect(getHelp(flagSchema)).toMatchSnapshot();

    flagSchema.content.deprecated = '--batman';
    expect(getHelp(flagSchema)).toMatchSnapshot();
  });
});
