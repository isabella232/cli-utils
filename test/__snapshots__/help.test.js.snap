// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help getHelp 1`] = `
"  --content            The path from which static content will be served
  --dev-ware           Set options for webpack-dev-middleware. e.g. --dev.publicPath /
  --help               Show usage information and the options listed here.
  --port               The port the app should listen on. Default: 8080
  --require, -r        Preload one or more modules before loading the webpack configuration
  --things, -batman    Where does he get such wonderful toys?
  --version            Display the webpack-command version
                       and a newline for great justice"
`;

exports[`help getHelp 2`] = `
"  --content            [1mDeprecated.[22m Please use --batman.
                       The path from which static content will be served
  --dev-ware           Set options for webpack-dev-middleware. e.g. --dev.publicPath /
  --help               Show usage information and the options listed here.
  --port               The port the app should listen on. Default: 8080
  --require, -r        Preload one or more modules before loading the webpack configuration
  --things, -batman    Where does he get such wonderful toys?
  --version            Display the webpack-command version
                       and a newline for great justice"
`;
